alembic/env.py

```diff
@@ -71,7 +73,35 @@ def run_migrations_online() -> None:
         context.configure(connection=connection, target_metadata=target_metadata)
 
         with context.begin_transaction():
-            context.run_migrations()
+            # Handle missing revisions gracefully
+            try:
+                context.run_migrations()
+            except Exception as e:
+                if "Can't locate revision identified by" in str(e):
+                    # Get the script directory to find the head revision for instructions
+                    script = ScriptDirectory.from_config(config)
+                    head_revision = script.get_current_head()
+                    
+                    logger = logging.getLogger("alembic.env")
+                    logger.error(f"Missing revision error encountered: {e}")
+                    logger.error("This typically happens when switching between branches with different migrations.")
+                    logger.error("The database contains a revision that doesn't exist in this branch.")
+                    logger.error("")
+                    logger.error("🔧 To fix this issue, run ONE of the following commands:")
+                    logger.error("1. alembic stamp head  # Recommended: stamps to current head revision")
+                    if head_revision:
+                        logger.error(f"2. python scripts/fix_alembic_revision.py  # Uses head revision: {head_revision}")
+                    logger.error("3. Reset your database if you don't have important data")
+                    logger.error("")
+                    logger.error("⚠️  WARNING: Do NOT manually update the alembic_version table unless you know what you're doing!")
+                    
+                    raise RuntimeError(
+                        "Alembic revision mismatch detected. "
+                        "Please run 'alembic stamp head' or 'python scripts/fix_alembic_revision.py' to fix this issue. "
```

Comment:
Our cli users won't have any access to our scripts/xxx.py file so we should never show such a message